lambdas/es/indexer/index.py

```diff
@@ -1,74 +1,183 @@
 """
-phone data into elastic for supported file extensions
+phone data into elastic for supported file extensions.
+note: we truncated inbound documents to no more than DOC_SIZE_LIMIT characters
+(this bounds memory pressure and request size to elastic)
 """
+
 from datetime import datetime
+from math import floor
 import json
 import os
-from urllib.parse import unquote
+from urllib.parse import unquote, unquote_plus
 
 from aws_requests_auth.aws_auth import AWSRequestsAuth
-import botocore
 import boto3
+from botocore.exceptions import ClientError
 from elasticsearch import Elasticsearch, RequestsHttpConnection
-from elasticsearch.exceptions import RequestError
+from elasticsearch.helpers import bulk
 import nbformat
-import tenacity
+from tenacity import stop_after_attempt, stop_after_delay, retry, wait_exponential
 
+# 10 MB, see https://amzn.to/2xJpngN
+CHUNK_LIMIT_BYTES = 10_000_000# 10MB
 DEFAULT_CONFIG = {
-    'to_index': [
-        '.ipynb',
-        '.md',
-        '.rmd',
+    "to_index": [
+        ".ipynb",
+        ".md",
+        ".rmd",
     ]
 }
-
+DOC_SIZE_LIMIT_BYTES = 10_000_000# 10MB
```

Comment:
Suggestion: add a parameter to flush the queue and send to elastic when the queue hits a certain size. That way the lambda will be able to process large batches without running out of memory.